src/NuGet.Clients/NuGet.PackageManagement.UI/Models/DetailControlModel.cs

```diff
@@ -509,8 +499,7 @@ public DetailedPackageMetadata PackageMetadata
                     PackageDeprecationReasons = newDeprecationReasons;
                     PackageDeprecationAlternatePackageText = newAlternatePackageText;
 
-                    IEnumerable<PackageVulnerabilityMetadataContextInfo> vulnerabilities = _packageMetadata?.Vulnerabilities;
-                    PackageVulnerabilities = vulnerabilities?.ToList();
+                    PackageVulnerabilities = _packageMetadata?.Vulnerabilities?.ToList();
```

Comment:
If we can use `.ToList()` instead of `.ToArrray()` here 
https://github.com/NuGet/NuGet.Client/pull/4175/files#diff-583146b1f5fc869123f4de17ff19759f66ba5edd3a7b8b6d43808a32955208e4R78 then we can change this line `502` to 
```cs
PackageVulnerabilities = _packageMetadata?.Vulnerabilities?.AsList();
// .ToList() changed to .AsList()
```
To use this extension method, I think we should edit `<IncludeNuGetSharedFiles>true</IncludeNuGetSharedFiles>` property to `NuGet.PackageManagement.UI.csproj` file.

https://github.com/NuGet/NuGet.Client/blob/dev/build/Shared/SharedExtensions.cs#L17

Reply:
@kartheekp-ms What problem are we solving by using AsList over ToList here?

I'm not quite understanding the first part of your comment either. You say to use ToList instead of ToArray but I'm not using ToArray. Could you clarify what you mean?

Reply:
https://github.com/NuGet/NuGet.Client/blob/966b1b21eec28834365a8604744dab682c8326ea/build/Shared/SharedExtensions.cs#L13-L14 comment has the information you are looking for. What `AsList()` extension method does is it avoids creating a new list if the underlying collection of the enumerable is of type `List`. Hope this helps.

Reply:
I want the list that is exposed in the ViewModel to explicitly be a new list, not exposing the underlying model directly. A ViewModel should not expose a list directly to the View. This is by design and after looking carefully at AsList, it is specifically not what I want this code to do since it will assign the instance of the PackageMetadata's Vulnerabilities list to the ViewModel's PackageVulnerabilities and expose it directly to the view.